src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1676,12 +1679,17 @@
                         (filter expired?)
                         sort)
         drop-one (fn [table]
-                   (update-lock-status status-key inc)
-                   (try!
-                    (jdbc/do-commands
-                     (format "drop table if exists %s cascade" table))
-                    (finally
-                      (update-lock-status status-key dec))))
+                   (let [{current :version} (sutils/db-metadata)
+                          reduced-locking pg14-db]
+                     (update-lock-status status-key inc)
+                     (try!
+                       (if-not (neg? (compare current reduced-locking))
+                         (jdbc/do-commands-outside-txn
```

Comment:
What happens here if there's an outer transaction in play...particularly in regards to the following drop table?